internal/controller/dscinitialization/servicemesh.go

```diff
@@ -0,0 +1,217 @@
+package dscinitialization
+
+import (
+	"context"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/api/dscinitialization/v1"
+	serviceApi "github.com/opendatahub-io/opendatahub-operator/v2/api/services/v1alpha1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/internal/controller/status"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/conditions"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
+)
+
+func (r *DSCInitializationReconciler) handleServiceMesh(ctx context.Context, dscInit *dsciv1.DSCInitialization) error {
+	log := logf.FromContext(ctx)
+
+	if dscInit.Spec.ServiceMesh == nil {
+		log.Info("ServiceMesh not defined in DSCI, deleting ServiceMesh CR if present")
+		return r.deleteServiceMesh(ctx)
+	}
+
+	conditions := dscInit.Status.Conditions
+	switch dscInit.Spec.ServiceMesh.ManagementState {
```

Comment:
Because for the `ServiceMesh` service we support `Managed` and `Unmanaged`, I'd recommend following the `Kueue` implementation where the `ManagementState` flags is passed also to the related CR, this allow for a consistent condition handling as it would make the `ServiceMesh` CR as the source of truth